mllib/src/main/scala/org/apache/spark/mllib/util/LinearDataGenerator.scala

```diff
@@ -125,6 +124,58 @@ object LinearDataGenerator {
   }
 
   /**
+   * @param intercept Data intercept
+   * @param weights  Weights to be applied.
+   * @param xMean the mean of the generated features. Lots of time, if the features are not properly
+   *              standardized, the algorithm with poor implementation will have difficulty
+   *              to converge.
+   * @param xVariance the variance of the generated features.
+   * @param nPoints Number of points in sample.
+   * @param seed Random seed
+   * @param eps Epsilon scaling factor.
+   * @return Seq of LabeledPoint includes sparse vectors..
+   */
```

Comment:
Yes, I also thought it is good idea. But `LinearDataGenerator` is used as static object, then we have to pass `sparsity` as parameter to `generateLinearInput`. This method seems to be used a lot of suites. It is necessary to change a lot of method reference. 
Therefore it might be better to do in separate JIRA. What do you thing about?